pkg/ccl/backupccl/backup_test.go

```diff
@@ -9269,6 +9270,43 @@ func TestExcludeDataFromBackupDoesNotHoldupGC(t *testing.T) {
 		})
 }
 
+// TestProtectRestoreSpans ensures the a protected timestamp is issued before
+// the restore flow begins and is released when the restore ends.
+func TestProtectRestoreSpans(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	numAccounts := 100
+	_, sqlDB, _, cleanupFn := backupRestoreTestSetupWithParams(t, 3 /* nodes */, numAccounts,
+		InitManualReplication, base.TestClusterArgs{})
+	defer cleanupFn()
+
+	sqlDB.Exec(t, `BACKUP INTO $1`, localFoo)
+
+	// Begin a Restore and assert that PTS with the correct target was persisted
+	sqlDB.Exec(t, `SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_flow'`)
+	var jobId jobspb.JobID
+	sqlDB.QueryRow(t, `RESTORE DATABASE data FROM LATEST IN $1 WITH detached, new_db_name=data2`, localFoo).Scan(&jobId)
+	jobutils.WaitForJobToPause(t, sqlDB, jobId)
+
+	restoreDetails := jobutils.GetJobPayload(t, sqlDB, jobId).GetRestore()
+	require.NotNil(t, restoreDetails.ProtectedTimestampRecord)
+
+	target := ptutil.GetPTSTarget(t, sqlDB, restoreDetails.ProtectedTimestampRecord)
+	targetIDs := target.GetSchemaObjects()
```

Comment:
occasionally `targetIDs.IDs[0]` causes a nil pointer exception, which seems weird. still investigating why.

Reply:
there was a bug in the code. Addressed.